src/gui/owncloudsetupwizard.cpp

```diff
@@ -128,15 +129,46 @@ void OwncloudSetupWizard::slotDetermineAuthType(const QString &urlString)
     account->setUrl(url);
     // Reset the proxy which might had been determined previously in ConnectionValidator::checkServerAndAuth()
     // when there was a previous account.
-    account->networkAccessManager()->setProxy(QNetworkProxy(QNetworkProxy::DefaultProxy));
+    account->networkAccessManager()->setProxy(QNetworkProxy(QNetworkProxy::NoProxy));
+
+    // Lookup system proxy in a thread https://github.com/owncloud/client/issues/2993
+    if (ClientProxy::isUsingSystemDefault()) {
+        qDebug() << "Trying to look up system proxy";
+        ClientProxy::lookupSystemProxyAsync(account->url(),
+                                            this, SLOT(slotSystemProxyLookupDone(QNetworkProxy)));
+    } else {
+        // We want to reset the QNAM proxy so that the global proxy settings are used (via ClientProxy settings)
+        account->networkAccessManager()->setProxy(QNetworkProxy(QNetworkProxy::DefaultProxy));
+        // use a queued invocation so we're as asynchronous as with the other code path
+        QMetaObject::invokeMethod(this, "slotContinueDetermineAuth", Qt::QueuedConnection);
+    }
```

Comment:
Optional: It'd be nice if this code wasn't duplicated with connectionvalidator.cpp somehow. A function that takes the two target slots? A minimal QObject that has two signals to connect to? It's still significant overhead for what is essentially four lines of code, but it might be worthwhile.